### DISCUSSION TITLE
Which is your favorite AI model for Perplexica?

### DISCUSSION THREAD

**Discussion by Zirgite** [ItzCrazyKns/Vane#312] [General]
I tried several ai models using Perplexica. Llama 3, Llama 3.1, Phi-3 Medium, Command-r 28b. 
My impressions are that even Llama 3 delivers good results, which means that even modest AI capable machines can benefit from this project using local models. 
Using Phi-3 Things are getting more interesting. The model is very good. All my tests where Perplexica rivals Perplexity copilot are using Phi-3 medium.
Command-r even if this model does not rank well in leaderboards (which do measure mainly raw logic capabilities) this is truly astonishing model that excels in creative writing and RAG. And the match with Perplexica is really good.  
To add further to the complexity different AI models perform in a different way with the same prompt, some deliver better results with simpler prompts some really seem to like detailed and markdown structured prompts. 
I did some tests using gpt4o as arbiter. The results are that even the smallest Llama 3.1 8B is having results that are not ranking very differently to the bigger models. Two conclusions either the search engine connection is a bottle neck, or for this task there is no need for a really big model. The second aligns with my constatations that for basic search not needing reasoning the vanilla perplexity does not perform much worse that copilot perplexity and on some occasions being on par with their results. 
As a whole I almost completely stopped using google and rely on perplexica and when I am not on my pc on perplexity. Especially if I search something like debugging.

**Comment by overcuriousity** [comment]
I look at it from 2 sides: local-only and cloud AI.

For cloud-based AI, I like claude3-haiku: Good, fast and dirt-cheap API (using it on a daily basis for 2 months, only cost under 2$ so far). (Unfortunately in the most recent version of Perplexica at the time of writing, There seem to be issues with rate-limiting)

For local-only setups i got the best results with the popular llama3.1:8b model - my hardware is restricted to a stone-age 4GB VRAM GPU. Phi3 was a miss, as it does not embed the citations correctly in many situations, along with some minor weaknesses. I also used the two larger gemma2 models with very impressive results and for sensitive topics, dolphin-llama3:8b did the trick as censorship is a thing with the stock llama models.

Interestingly, the results are heavily impacted by the embeddings model selected - subjectively even more than with the GPT model. The by far best results i had were with nomic-embed-text served by ollama. I would recommend the latter, as performance is not heavily affected.

**Reply by Zirgite** [reply]
It appears true that the embeddings do change the results, It is good to see someone to share the same as I was not entirely sure about it, I was thinking that I was hallucinating myself lol. 
The biggest advantage of perplexica it is not limited to consummer grade research as the other engines are. All the commercial search ai engines are **implicitely tuned to  deliver easy to understand results** and not to dig in seriously in the topic. But of course you can fix that telling them using the proper prompts who you are (student, researcher etc.) and what is your level of detail you need. 
As a result my conclusion is to get the most of this new tool it is better not to treat them as google but to get into real discussion of the topic, in that way you can finetune the research.

**Reply by overcuriousity** [reply]
Yeah, you are right - but i indeed would consider it as a replacement for Google etc. if you want to research any topic which requires reading through many search results.
I think the biggest advantage of Perplexity.ai-like applications is the possibility to save you time by summarizing all the search results into a single, high-quality answer. Perplexica is even better than commercial solutions, as it combines this with a powerful meta-search-engine like searxng, increasing privacy in the process with optional local AI-processing for maximum privacy.

Where it does not benefit much is when I search for specific code snippets for a specific problem, for that i stick either with stackoverflow or a good commercial GPT like claude.

I hope this project is not abandoned soon, as the development seemed to slow down a bit.

**Comment by merodeador** [comment]
I've been using Mistral-Nemo 12B with Perplexica and the results are great.

**Comment by vorktanamobay** [comment]
I agree, mistral-nemo 12b seems to be the best, at least for my 2080ti. q4_K_M is ok most of the time for me. What are you guys using for num_ctx on the chat model? The stock ollama 2048? I played around with 4096 and 8192, seems to help but really eats vram. I have a 2080ti so I can set num_ctx to 8k with q4_K_M and still get moderately fast speeds. Bumping down num_ctx to 4k with q4_K_M makes it fit in gpu vram and its pretty fast. Anyone else try messing with num_ctx? Is 8k really needed? I'm not sure. I also bumped temperature to 0.7 since I noticed in the code that is what the cloud models are using.

One other thing regarding embedding model, I also used nomic-embed-text:latest but I bumped its num_ctx to 8k, not sure if its needed but that's the max it can handle. Seems to work ok for me.

Update: I did some more testing and q5_K_M with 4096 num_ctx seems to be the sweet spot for response quality.

**Comment by bpawnzZ** [comment]
i'm using litellm connected to openrouter using openai/gpt-4o-mini and it work great. running small open embedding model

**Reply by TomLucidor** [reply]
How did you do that?

**Comment by yieme** [comment]
How did you get Perplexica to change it's default model or to connect to litellm? I see it's using langchain and I tried setting OPENAI_API_KEY, OPENAI_BASE_URL and OPENAI_MODEL_NAME as environment variables but they didn't seem to pick up.

**Comment by theronypony** [comment]
Qwen3-30b-A3b with a /no_think system prompt is fire

**Reply by BradKML** [reply]
Is it that good now? What bout DeepSeek distillation?

**Reply by djmaze** [reply]
For me, `hf.co/Qwen/Qwen3-14B-GGUF:Q6_K` is totally sufficient.

Combine that with `hf.co/nomic-ai/nomic-embed-text-v2-moe-gguf:Q6_K` for a good ollama-based multilingual embedding model.

**Reply by BradKML** [reply]
Is it the same for agentic code generation as well? Or some other size of Qwen3/DeepSeek model?

**Reply by delfireinoso** [reply]
Depends on you memory but Qwen3 or Gemma3 are great performers on Perplexica

**Reply by BradKML** [reply]
Let's say sometimes desktop with 16/32/64GB RAM, and sometimes OpenRouter but hoping for cheap providers

**Reply by delfireinoso** [reply]
I use a Mac with 32 Gb and if I am
on lan I can use a 24 Gb model as Gemma 30 b or Qwen 32

**Reply by djmaze** [reply]
> Is it the same for agentic code generation as well? Or some other size of Qwen3/DeepSeek model?

Not sure if you mean me? Did not try code generation with Perplexica, only using it for web (re)search.

Heck, for web search, it even works okay with `qwen3:8b` on my RTX 4070 laptop! That's the good thing about RAG tasks, you do not need big models at all.

**Reply by BradKML** [reply]
@djmaze I meant how models are often "dual use" between code generation and web search... maybe that is a bit of a bad idea IDK?

**Reply by TaoVonQi** [reply]
> For me, `hf.co/Qwen/Qwen3-14B-GGUF:Q6_K` is totally sufficient.
> 
> Combine that with `hf.co/nomic-ai/nomic-embed-text-v2-moe-gguf:Q6_K` for a good ollama-based multilingual embedding model.

How were you able to configure it with a custom embedding model?

**Reply by djmaze** [reply]
> > For me, `hf.co/Qwen/Qwen3-14B-GGUF:Q6_K` is totally sufficient.
> > Combine that with `hf.co/nomic-ai/nomic-embed-text-v2-moe-gguf:Q6_K` for a good ollama-based multilingual embedding model.
> 
> How were you able to configure it with a custom embedding model?

It's in the settings UI:

<img width="802" height="404" alt="image" src="https://github.com/user-attachments/assets/cb82ac39-f69b-45a1-9336-961a2f931c99" />

**Reply by TaoVonQi** [reply]
> > > For me, `hf.co/Qwen/Qwen3-14B-GGUF:Q6_K` is totally sufficient.
> > > Combine that with `hf.co/nomic-ai/nomic-embed-text-v2-moe-gguf:Q6_K` for a good ollama-based multilingual embedding model.
> > 
> > 
> > How were you able to configure it with a custom embedding model?
> 
> It's in the settings UI:
> <img alt="image" width="802" height="404" src="https://private-user-images.githubusercontent.com/7229/489196408-cb82ac39-f69b-45a1-9336-961a2f931c99.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTc3ODk5MzMsIm5iZiI6MTc1Nzc4OTYzMywicGF0aCI6Ii83MjI5LzQ4OTE5NjQwOC1jYjgyYWMzOS1mNjliLTQ1YTEtOTMzNi05NjFhMmY5MzFjOTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDkxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA5MTNUMTg1MzUzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjJkMzA0ZTNhMWNiOWJjYmVkNDA4ZGE3OTZiMDU5NTY0YTI2Yzc5YjBmYmM3MWU5NDcwODdiZjRmMDlmOTY2NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.SFUj73U6WllzXHoIwJ2PJeD0OQb9jt_jylXDzrdiDKk">

That's strange, my settings are hard-coded to hugging face with 3 preset models to choose from. Are you using a certain branch of the repo? I just checked out the main branch and built the image locally

**Reply by TaoVonQi** [reply]
Oh I guess it behaves differently when you specify the custom openai in the config.toml instead of ollama. 

I think this might be the reason I finally pull the trigger on ollama until a custom openai compatible model is supported for embedding models.

**Reply by djmaze** [reply]
Oh, yes, also I am running on #751.

**Comment by silverflight22** [comment]
Will you ever develop a custom API for Perplexica?????

**Comment by debernardis** [comment]
I am settling, for my M4 16GB, on https://huggingface.co/HauhauCS/Gemma-4-E4B-Uncensored-HauhauCS-Aggressive, an uncensored model, which fits, and works well without stalling when the context window is enlarged to the maximum. Not exceedingly quick but does it.

**Reply by delfireinoso** [reply]
Wich LLM manager are you using?

**Reply by debernardis** [reply]
LM Studio, which makes easy to change model parameters

**Reply by debernardis** [reply]
now testing gemma-4-e4b-it-8bit under oMLX - uses less memory and is quicker

**Reply by overcuriousity** [reply]
How does this run on 16 GiB VRAM with any reasonable context size (k/v-cache?)

**Comment by ktneely** [comment]
I find that the newly released qwen3.5 models work really well for
Perplexica and the 9b fits on reasonable hardware. Running on ollama with
parameters to widen the context window.

Qwen3.5 can spend a lot of time reasoning with itself, which will lengthen
the time before you get results.

**Reply by theronypony** [reply]
Thinking can be disabled for the qwen 3.5 models (or any reasoning model) in llama.cpp using this flag

`"chat_template_kwargs": {"enable_thinking": false}`

**Comment by debernardis** [comment]
Yep I also found qwen to be mumbling with itself for too much time, and
even with fiddling with parameters I was unable to stop it overthinking.


Ernesto de Bernardis
medico farmacologo
direttore ff UOC dipendenze patologiche ASP SR
orcid 0000-0002-6809-4096

Il Sab 4 Apr 2026, 23:48 Kevin ***@***.***> ha scritto:

> I find that the newly released qwen3.5 models work really well for
> Perplexica and the 9b fits on reasonable hardware. Running on ollama with
> parameters to widen the context window.
>
> Qwen3.5 can spend a lot of time reasoning with itself, which will lengthen
> the time before you get results.
>
>
